s3/src/main/scala/geotrellis/store/s3/S3LayerDeleter.scala

```diff
@@ -25,11 +25,15 @@ import org.log4s._
 import scala.collection.JavaConverters._
 
 class S3LayerDeleter(
-  val attributeStore: AttributeStore,
-  s3Client: => S3Client
-) extends LayerDeleter[LayerId] {
+      val attributeStore: AttributeStore,
+      s3Client: => S3Client,
+      val requestLimit: Int = 1000
+    ) extends LayerDeleter[LayerId] {
   @transient private[this] lazy val logger = getLogger
 
+  def apply(attributeStore: AttributeStore, s3Client: => S3Client, requestLimit: Int): S3LayerDeleter =
```

Comment:
Wow, would it even work? This constructor duplicated the original one.

Could you move it into the companion object? ([here](https://github.com/locationtech/geotrellis/blob/cfa3963593577c8564855695872d429ae69d287f/s3/src/main/scala/geotrellis/store/s3/S3LayerDeleter.scala#L80))

The idea of this move is to make API usage a bit nicer:

```scala
// class constructor usage
val deleter = new S3LayerDeleter(attributeStore, s3Client, requestLimit)
// companion object apply usage
val deleter S3LayerDeleter(attributeStore, s3Client, requestLimit)
```

Reply:
Sorry about that. Let me know if it is ok now.

Reply:
🚀